src/Core/AccurateComparison.h

```diff
@@ -515,15 +515,24 @@ inline bool NO_SANITIZE_UNDEFINED convertNumeric(From value, To & result)
         return true;
     }
 
-    /// Note that NaNs doesn't compare equal to anything, but they are still in range of any Float type.
-    if (isNaN(value) && std::is_floating_point_v<To>)
+    if constexpr (std::is_floating_point_v<From> && std::is_floating_point_v<To>)
     {
-        result = value;
-        return true;
+        /// Note that NaNs doesn't compare equal to anything, but they are still in range of any Float type.
+        if (isNaN(value))
+        {
+            result = value;
+            return true;
+        }
+    }
+
+    if (accurate::greaterOp(value, std::numeric_limits<To>::max())
```

Comment:
What if we convert Float64's inf to Float32's inf or vice versa?

Reply:
Also https://en.cppreference.com/w/cpp/types/numeric_limits/lowest ?

Reply:
Will investigate floating point types conversion, seems like there is are a bunch of problems.

Reply:
Related to infinite conversions we can add special case above in addition to isNan.

Reply:
Ok.

Reply:
BTW, loss of precision is Ok. For example,

`toFloat32(0.1) IN (SELECT 0.1)` should work.

Reply:
So, we can assume all the range of [lowest, max] of Float32 in Float64 to be representable in Float32.

Reply:
(actually I'm not 100% sure about it)

Reply:
Updated. As discussed does not create specialization for case like ```toFloat32(0.1) IN (SELECT 0.1)```.